tests/devices/test_default_qutrit.py

```diff
@@ -328,6 +599,71 @@ def circuit(mat):
         assert np.allclose(state, expected_out, atol=tol)
 
 
+class TestTensorExpval:
+    """Test tensor expectation values"""
+
+    # TODO: Add tests for non-parametric observables and identity
+
+    def test_hermitian_hermitian(self, tol):
+        """Test that a tensor product involving two Hermitian matrices works correctly"""
+        dev = qml.device("default.qutrit", wires=3)
+
+        A1 = np.array([[3, 0, 0], [0, -3, 0], [0, 0, -2]])
+
+        A = np.array([[1, 0, 0], [0, -1, 0], [0, 0, 2]])
+        B = np.array([[4, 0, 0], [0, -2, 0], [0, 0, 1]])
+        A2 = np.kron(A, B)
+
+        obs = qml.THermitian(A1, wires=[0]) @ qml.THermitian(A2, wires=[1, 2])
+
+        dev.apply(
+            [
+                qml.QutritUnitary(U_shift, wires=0),
+                qml.QutritUnitary(U_tadd, wires=[0, 1]),
+                qml.QutritUnitary(U_shift, wires=0),
+            ],
+            obs.diagonalizing_gates(),
+        )
+
+        res = dev.expval(obs)
+
+        expected = 8
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_hermitian_two_wires_identity_expectation(self, tol):
+        """Test that a tensor product involving an Hermitian matrix for two wires and the identity works correctly"""
+        dev = qml.device("default.qutrit", wires=3)
+
+        A = np.array([[-2, 0, 0], [0, 8, 0], [0, 0, -1]])
+        Identity = np.eye(3)
+        H = np.kron(np.kron(Identity, Identity), A)
+        obs = qml.THermitian(H, wires=[2, 1, 0])
+
+        dev.apply(
+            [
+                qml.QutritUnitary(U_thadamard_01, wires=0),
+                qml.QutritUnitary(U_shift, wires=0),
+                qml.QutritUnitary(U_shift, wires=1),
+                qml.QutritUnitary(U_tadd, wires=[0, 1]),
+            ],
+            obs.diagonalizing_gates(),
+        )
+        res = dev.expval(obs)
+
+        expected = 3.5 * 1 * 1
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+
+# TODO: Add tests for tensor non-parametrized observables
+class TestTensorVar:
```

Comment:
Why is it not possible to add these yet?

Reply:
It is definitely possible to add them. I've added tests for tensor variance in a future PR and I think it's a good idea to leave it that way for now, purely because this PR is already almost 1000 additions.